apps/nuxt3/app.vue

```diff
@@ -1,55 +1,55 @@
 <template>
-    <div>
-        <NuxtWelcome />
-        <vue-particles
-            id="tsparticles"
-            :options="options"
-            :particles-init="particlesInit"
-        />
-    </div>
+  <div>
+    <NuxtWelcome />
+    <vue-particles
+      id="tsparticles"
+      :options="options"
+      :particles-init="particlesInit"
+      @particles-loaded="particlesLoaded"
+    />
+  </div>
 </template>
-<script>
+<script setup lang="ts">
 import { loadFull } from "tsparticles";
+import type { ISourceOptions, Container } from "tsparticles-engine";
 
-export default {
-    name: "NuxtTutorial",
-    data() {
-        return {
-            options: {
-                background: {
-                    color: "#fff"
-                },
-                fullScreen: {
-                    enable: true,
-                    zIndex: -1
-                },
-                particles: {
-                    color: {
-                        value: "#000"
-                    },
-                    links: {
-                        color: "#000",
-                        enable: true
-                    },
-                    move: {
-                        enable: true
-                    },
-                    number: {
-                        value: 100
-                    }
-                }
-            }
-        };
+const options = {
+  background: {
+    color: "#fff"
+  },
+  fullScreen: {
+    enable: true,
+    zIndex: -1
+  },
+  particles: {
+    color: {
+      value: "#000"
     },
-    methods: {
-        particlesInit: async (engine) => {
-            await loadFull(engine);
-        }
+    links: {
+      color: "#000",
+      enable: true
+    },
+    move: {
+      enable: true
+    },
+    number: {
+      value: 100
     }
-};
+  }
+} as ISourceOptions
+
+const particlesInit = async (engine) => {
+  await loadFull(engine)
+  console.log(engine)
+}
+
+const particlesLoaded = (container: Container) => {
+  console.log(toRaw(container))
```

Comment:
In the Vue3 app this `toRaw` is not used, is this something related to Nuxt?

Reply:
I just did this so that the object is printed to the console without Vue's proxy attached; it makes the log a bit cleaner.

Reply:
Ok, is it missing in the Vue3 app or is it necessary only in Nuxt3 app?

Reply:
Missing in the Vite one. It doesn't make too much of a difference, though.